### PR TITLE
feat(unlock-app, ui) - Add Card component and refactor UI

### PR DESCRIPTION
**Pull request by kalidiagne**
# Description

- [x] Add `Card` component to have a common styles for cards across the app 
- [x] Back button on metadata page redirect to previous page 
- [x] Allow custom image + redirect URL for `AppLayout` so we can customize the header logo 
- [x] Show open sea URL on keychain only when user is logged 

**Card with shadow** 

Screenshot 2023-04-13 at 13 50 19

**Card danger** 
Screenshot 2023-04-13 at 13 50 13

**Card secondary**
 
Screenshot 2023-04-13 at 13 50 10

**Card primary**

Screenshot 2023-04-13 at 13 50 06

**Events by Unlock logo on certificate page** 

Screenshot 2023-04-13 at 13 41 24

## Use of `Card` component in app 

**Locks page** 

Screenshot 2023-04-13 at 13 24 53

**Keychain** 

Screenshot 2023-04-13 at 13 18 47

**Manage lock page** 

Screenshot 2023-04-13 at 12 53 43

Screenshot 2023-04-13 at 02 28 22

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: packages/ui/lib/components/Card/Card.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  description?: string
+}
+export const CardLabel = ({ title, description }: CardLabelProps) => {
+  return (
+    <div className="flex flex-col gap-2 md:col-span-2">
+      <CardTitle>{title}</CardTitle>
+      {description && <CardDescription>{description}</CardDescription>}
+    </div>
+  )
+}
```
Comment (searchableguy) [inline packages/ui/lib/components/Card/Card.tsx:65]: One pattern you can use here to simplify and make it easier to use as a user.

Card.Title = CardTitle

so in actual usage, one could do 
```jsx
<Card>
   <Card.Title> hello </Card.Title>
</Card>
```

Does that make sense?

File: packages/ui/lib/components/Card/Card.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+export const Card = ({
+  variant,
+  shadow,
+  padding,
+  children,
+  className,
+}: CardBaseProps) => {
+  return (
```
Comment (searchableguy) [inline packages/ui/lib/components/Card/Card.tsx:42]: I wonder if you can make this polygot component so someone can pass as="div" to change what gets rendered with styles. WDYT?

**Review by searchableguy — APPROVED** [review]
LGTM. Two minor enhancement if you want to add.

File: packages/ui/lib/components/Card/Card.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    padding: 'sm',
+  },
+})
+
+const CardTitle = classed.span('text-xl font-bold text-brand-ui-primary')
+const CardDescription = classed.span('text-base text-brand-dark"')
+
+type CardBaseProps = React.ComponentProps<typeof CardBase> & {
+  icon?: React.ReactNode
+  as?: ReactTag
```
Comment (searchableguy) [inline packages/ui/lib/components/Card/Card.tsx:34]: This is not correct type. It's also internal to headless so we should not use

Comment (kalidiagne) [reply packages/ui/lib/components/Card/Card.tsx]: fixed ;)

**Pull request merged**